### ISSUE TITLE
MPU9250_BasicAHRS with MPUTeapot 

### ISSUE THREAD

**Issue by rkuo2000** [kriswiner/MPU-9250#301]
Hi,
I have modified MPU9250_BasicAHRS to support MPUTeapot
my code is in https://github.com/rkuo2000/arduino/tree/master/examples/MPU9250_BasicAHRS

I don't connect to LCD, so I removed LCD display part, and added teapot packet Serial.write ( line #427)
#ifdef OUTPUT_TEAPOT // for Teapot Packet
    // display quaternion values in InvenSense Teapot demo format:         
    teapotPacket[2] = int(q[0] * 16384) >>8;
    teapotPacket[3] = int(q[0] * 16384) & 0x00ff;
    teapotPacket[4] = int(q[1] * 16384) >> 8;
    teapotPacket[5] = int(q[1] * 16384) & 0x00ff;
    teapotPacket[6] = int(q[2] * 16384) >> 8;
    teapotPacket[7] = int(q[2] * 16384) & 0x00ff;
    teapotPacket[8] = int(q[3] * 16384) >> 8;
    teapotPacket[9] = int(q[3] * 16384) & 0x00ff;
    Serial.write(teapotPacket, 14);  	
    teapotPacket[11]++; // packetCount, loops at 0xFF on purpose                     
#else  // for Teapot Packet
    if(SerialDebug) {
...

The MPUTeapot can display 3D plane rotating according to the quaternions,
but it seems the yaw will be drifting, the quaternions print out show it is dripping too
(both MPUTeapot.pde & MPU9250_BasicAHRS.ino)

But when I use MPU9250 DMP quaternions, it is not drifting and very stable.
https://github.com/rkuo2000/arduino/tree/master/examples/MPU9250_DMP_AHRS
Could you tell where the problem could be ?

Richard

**Comment by kriswiner** [comment]
No, but how did you calibrate your sensors?

**Comment by rkuo2000** [comment]
I did not modify calibrateMPU9250, let me look into the calibration, thanks for the direction :D

**Comment by kriswiner** [comment]
If you are using the old basic code it doesn't do magnetometer calibration,
so you will need to add this to get any kind of accuracy and stability.

**Comment by rkuo2000** [comment]
Yes, after adding magcalMPU9250, just a little drifting shown on Teapot now :D
For the little yaw drifting, maybe I did not do it well in figure-8 ?
I test it with a magnet, it looks immune to me. 

My modified code is based on MPU9250_MS5637_AHRS_t3.ino 
MS5637 and Interrupt are commented, and added Teapot packet
https://github.com/rkuo2000/arduino/tree/master/examples/MPU9250_BasicAHRS2

**Comment by kriswiner** [comment]
I would run the sensors (at least the gyro) at 500 or 1000 Hz sample rate
and make sure your MCU is able to achieve at least 4 kHz fusion rate for
best results. Of course, you need to make sure the accel and gyro bias
offsets are corrected as well.

**Comment by rkuo2000** [comment]
I will look into those bias and calibration, thanks !
